src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -209,7 +261,83 @@ public static void Unregister()
                 throw new InvalidOperationException(error);
             }
 
+#if NET46
             AppDomain.CurrentDomain.AssemblyResolve -= registeredHandler;
+#else
+            AssemblyLoadContext.Default.Resolving -= registeredHandler;
+#endif
+        }
+
+        /// <summary>
+        ///     Ensures the proper MSBuild environment variables are populated for DotNet SDK.
+        /// </summary>
+        /// <param name="msbuildPath">
+        ///     Path to the directory containing the DotNet SDK.
+        /// </param>
+        private static void ApplyDotNetSdkEnvironmentVariables(string dotNetSdkPath)
+        {
+            const string MSBUILD_EXE_PATH = nameof(MSBUILD_EXE_PATH);
+            const string MSBuildExtensionsPath = nameof(MSBuildExtensionsPath);
+            const string MSBuildSDKsPath = nameof(MSBuildSDKsPath);
+
+            var variables = new Dictionary<string, string>
+            {
+                [MSBUILD_EXE_PATH] = dotNetSdkPath + "MSBuild.dll",
+                [MSBuildExtensionsPath] = dotNetSdkPath,
+                [MSBuildSDKsPath] = dotNetSdkPath + "Sdks"
```

Comment:
Some tasks will also expect DOTNET_HOST_PATH to be set on Core to the path to dotnet[.exe]

Reply:
This is not needed since you rely on dotnet being on PATH.